internal/impl/babysitter.go

```diff
@@ -176,8 +183,18 @@ func (b *babysitter) watchPods(ctx context.Context, component string) error {
 		return fmt.Errorf("unable to determine group name for component %s", component)
 	}
 	name := deploymentName(b.app.Name, rs, b.cfg.DeploymentId)
+	//timeOut := int64(60)
+	// , TimeoutSeconds: &timeOut
 	opts := metav1.ListOptions{LabelSelector: fmt.Sprintf("serviceweaver/name=%s", name)}
-	watcher, err := b.clientset.CoreV1().Pods(b.cfg.Namespace).Watch(ctx, opts)
+
+	// watcher factory
+	watchFunc := func(options metav1.ListOptions) (watch.Interface, error) {
```

Comment:
can we inline this?

`
	watcher, err := watch2.NewRetryWatcher("1", &cache.ListWatch{
		WatchFunc: func(metav1.ListOptions) (watch.Interface, error) {
			w, e := b.clientset.CoreV1().Pods(b.cfg.Namespace).Watch(ctx, opts)
			return w, e
		},
	})
`

Reply:
done